tonic-web/src/lib.rs

```diff
@@ -87,49 +87,58 @@
 #![doc(html_root_url = "https://docs.rs/tonic-web/0.4.0")]
 #![doc(issue_tracker_base_url = "https://github.com/hyperium/tonic/issues/")]
 
-pub use config::Config;
+pub use layer::GrpcWebLayer;
+pub use service::GrpcWebService;
 
 mod call;
-mod config;
-mod cors;
+mod layer;
 mod service;
 
-use crate::service::GrpcWeb;
+use http::header::HeaderName;
 use std::future::Future;
 use std::pin::Pin;
+use std::time::Duration;
 use tonic::body::BoxBody;
-use tonic::transport::NamedService;
+use tower_http::cors::{AllowOrigin, Cors, CorsLayer};
+use tower_layer::Layer;
 use tower_service::Service;
 
-/// enable a tonic service to handle grpc-web requests with the default configuration.
+const DEFAULT_MAX_AGE: Duration = Duration::from_secs(24 * 60 * 60);
+const DEFAULT_EXPOSED_HEADERS: [&str; 3] =
+    ["grpc-status", "grpc-message", "grpc-status-details-bin"];
+const DEFAULT_ALLOW_HEADERS: [&str; 4] =
+    ["x-grpc-web", "content-type", "x-user-agent", "grpc-timeout"];
```

Comment:
Perhaps we could make these `pub` for people instantiating the `CorsLayer` manually?

Reply:
Is there a big benefit here? I personally like to stay on the conservative side of making things public just incase you need to change them in the future.